### PR TITLE
8282651: ZGC: vmTestbase/gc/ArrayJuggle/ tests fails intermittently with exit code 97

### PR DESCRIPTION
**Pull request by rsunderbabu**
In systems with many cores, lots of threads create arrays very quickly and result in OOM. The intention of the ArrayJuggle tests is not to stress gc but to do array juggling. Hence number of objects juggling can be restricted to a minimum so that we don't face OOM even when threads available are high.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8282651](https://bugs.openjdk.org/browse/JDK-8282651): ZGC: vmTestbase/gc/ArrayJuggle/ tests fails intermittently with exit code 97

### Reviewers
 * [Leonid Mesnik](https://openjdk.org/census#lmesnik) (@lmesnik - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/11830/head:pull/11830` \
`$ git checkout pull/11830`

Update a local copy of the PR: \
`$ git checkout pull/11830` \
`$ git pull https://git.openjdk.org/jdk pull/11830/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 11830`

View PR using the GUI difftool: \
`$ git pr show -t 11830`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/11830.diff">https://git.openjdk.org/jdk/pull/11830.diff</a>

</details>

### CODE REVIEW

File: test/hotspot/jtreg/vmTestbase/gc/ArrayJuggle/Juggle01/Juggle01.java [hunk 1]
```diff
@@ -67,7 +72,9 @@ public void run() {
                 log.debug("Garbage producer: " + garbageProducer);
                 log.debug("Memory strategy: " + memoryStrategy);
                 long memory = runParams.getTestMemory();
-                int objectCount = memoryStrategy.getCount(memory);
+                // Keep object count to 4, intention is to juggle
+                // not to overwhelm gc
+                int objectCount = 4;
```
Comment (lmesnik) [inline test/hotspot/jtreg/vmTestbase/gc/ArrayJuggle/Juggle01/Juggle01.java:77]: I don't see how it is used.

Comment (rsunderbabu) [reply test/hotspot/jtreg/vmTestbase/gc/ArrayJuggle/Juggle01/Juggle01.java:77]: array = new Object[objectCount - 1];
This is the array where juggling happens. Limiting the object count limits the number of objects created.

**Comment by stefank** [comment]
> The intention of the ArrayJuggle tests is not to stress gc but to do array juggling.

FWIW, the ArrayJuggle tests are very good at stressing the GC and finding bugs. Is this change going to make this a worse GC stress test?

**Comment by rsunderbabu** [comment]
> 

I am abandoning that approach. GC stressing would be retained.

**Review by lmesnik — APPROVED** [review]

**Review by lmesnik — APPROVED** [review]

**Pull request closed without merging**